bundles/org.openhab.voice.whisperstt/src/main/java/org/openhab/voice/whisperstt/internal/WhisperSTTService.java

```diff
@@ -0,0 +1,654 @@
+                        audioSamplesOffset = 0;
+                        logger.debug("Partial transcription: {}", tempTranscription);
+                        logger.debug("Transcription: {}", transcription);
+                    }
+                } finally {
+                    if (releaseContext) {
+                        ctx.close();
+                    }
+                }
+                // emit result
+                if (!aborted.get()) {
+                    sttListener.sttEventReceived(new RecognitionStopEvent());
+                    logger.debug("Final transcription: '{}'", transcription);
+                    if (!transcription.isBlank()) {
+                        sttListener.sttEventReceived(new SpeechRecognitionEvent(transcription.trim(), 1));
+                    } else {
+                        emitSpeechRecognitionError(sttListener);
+                    }
+                }
+            } catch (IOException e) {
+                logger.warn("Error running speech to text: {}", e.getMessage());
+                emitSpeechRecognitionError(sttListener);
+            } catch (UnsatisfiedLinkError e) {
+                logger.warn("Missing native dependency: {}", e.getMessage());
+                emitSpeechRecognitionError(sttListener);
+            }
+        });
+    }
+
+    private WhisperFullParams getWhisperFullParams(WhisperContext context, Locale locale) throws IOException {
+        WhisperSamplingStrategy strategy = WhisperSamplingStrategy.valueOf(config.samplingStrategy);
+        var params = new WhisperFullParams(strategy);
+        params.temperature = config.temperature;
+        params.nThreads = config.threads;
+        params.audioCtx = config.audioContext;
+        params.speedUp = config.speedUp;
+        params.beamSearchBeamSize = config.beamSize;
+        params.greedyBestOf = config.greedyBestOf;
+        if (!config.initialPrompt.isBlank()) {
+            params.initialPrompt = config.initialPrompt;
+        }
+        if (grammar != null) {
+            params.grammar = grammar;
+            params.grammarPenalty = config.grammarPenalty;
+        }
+        // there is no single language models other than the english ones
+        params.language = getWhisper().isMultilingual(context) ? locale.getLanguage() : "en";
+        // implementation assumes this options
+        params.translate = false;
+        params.detectLanguage = false;
+        params.printProgress = false;
+        params.noTimestamps = true;
+        params.printRealtime = false;
+        params.printSpecial = false;
+        params.printTimestamps = false;
+        params.suppressBlank = true;
+        params.suppressNonSpeechTokens = true;
+        params.singleSegment = true;
+        params.noContext = true;
+        return params;
+    }
+
+    private void emitSpeechRecognitionError(STTListener sttListener) {
+        String errorMessage = !config.errorMessage.isBlank() ? config.errorMessage : "Sorry, something went wrong";
```

Comment:
One solution would be to let the blank message in case the user defined it blank.
In the dialog processor, if the message is blank, nothing is said.
Like that the user has the choice between having a feedback or not when the STT service is not able to proceed.
WDYT ?

Reply:
Yes, makes sense. I'll go with that.

Reply:
I have created a noResultsMessage configuration to use on the empty transcription case like in the voskstt add-on.